src/lib/DefaultRuleManager.ts

```diff
@@ -143,6 +144,31 @@ export class DefaultRuleManager implements RuleManager {
 		}
 	}
 
+	protected async emitRunTelemetry(runDescriptorList: RunDescriptor[]): Promise<void> {
+		const runTelemetryObject: TelemetryData = {
+			eventName: 'ENGINE_EXECUTION'
+		};
+
+		const executedEngineNames: Set<string> = new Set(runDescriptorList.map(d => d.engine.getName().toLowerCase()));
+
+		const allEngines: RuleEngine[] = await Controller.getAllEngines();
+		for (const engine of allEngines) {
+			const engineName = engine.getName().toLowerCase();
+			// Instead of concatenating the engines into a string, we assign each engine to a boolean value based on
+			// whether it's being executed. Theoretically, that will be more useful for gathering meaningful metrics.
```

Comment:
Is there a way to confirm this theory? Also, if a single run invoked multiple engines, I wonder how that would account for our numbers.

Reply:
I'm not sure how we'd confirm this without actually looking at the telemetry org and seeing what sort of analysis we can do with that.
But theoretically, this should at least allow us to easily identify what percentage of runs invoke which engines, which is the single most important piece of data IMO.

Reply:
Makes sense. In addition to this, would it be worth capturing how many engines are run per invocation? Possibly store the comma-separated string of engines as well. This way, we'll know which combination of engines are mostly invoked together.

Reply:
If we sort the engine name list, the grouping of strings would be consistent as well.